src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,193 @@
 package core.basesyntax;
 
+import java.util.Iterator;
+import java.util.List;
+import java.util.ListIterator;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] elements;
+    private int size;
+
+    public ArrayList() {
+        this.elements = new Object[DEFAULT_CAPACITY];
+        this.size = 0;
+    }
+
+    private void ensureCapacity() {
+        if (size == elements.length) {
+            int newCapacity = (int) (elements.length * 1.5 + 1);
+            Object[] newArray = new Object[newCapacity];
+            System.arraycopy(elements, 0, newArray, 0, elements.length);
+            elements = newArray;
+        }
+    }
+
     @Override
-    public void add(T value) {
+    public boolean add(T element) {
+        ensureCapacity();
+        elements[size++] = element;
+        return true;
+    }
 
+    @Override
+    public T get(int index) {
+        if (index < 0 || index >= size) {
+            throw new ArrayListIndexOutOfBoundsException(
+                    "Index " + index + " to go beyond the list");
+        }
+        return (T) elements[index];
     }
 
     @Override
-    public void add(T value, int index) {
+    public T remove(int index) {
+        if (index < 0 || index >= size) {
+            throw new ArrayListIndexOutOfBoundsException(
```

Comment:
Make your exceptions informative. The exception message should clearly state the problem. For example, 'Index ' + index + ' is out of bounds for size ' + size.